packages/block-library/src/gallery/utils.js

```diff
@@ -36,17 +40,32 @@ export function getHrefAndDestination(
 			return {
 				href: image?.source_url || image?.url, // eslint-disable-line camelcase
 				linkDestination: IMAGE_LINK_DESTINATION_MEDIA,
+				lightbox: lightboxSetting?.enabled
+					? { enabled: false }
+					: undefined,
 			};
 		case LINK_DESTINATION_ATTACHMENT_WP_CORE:
 		case LINK_DESTINATION_ATTACHMENT:
 			return {
 				href: image?.link,
 				linkDestination: IMAGE_LINK_DESTINATION_ATTACHMENT,
+				lightbox: lightboxSetting?.enabled
+					? { enabled: false }
+					: undefined,
+			};
+		case LINK_DESTINATION_LIGHTBOX:
+			return {
+				href: undefined,
+				lightbox: ! lightboxSetting?.enabled
+					? { enabled: true }
+					: undefined,
```

Comment:
I think this control in the Gallery block will _always_ apply the selected option to the Image blocks inside, so there is no need to check if the Image block's lightbox is enabled by default.

Additionally, the Image block only has the `enabled` property in the `lightbox` attribute, but if other properties are added in the future, `{enabled: true}` may remove the other properties.

Therefore, we may need to change the arguments of the `getHrefAndDestination` function to something like this:

```javascript
// edit.js
function setLinkTo( value ) {
	// ...
	getBlock( clientId ).innerBlocks.forEach( ( block ) => {
		// ...
		changedAttributes[ block.clientId ] = getHrefAndDestination(
			image,
			value,
			false,
			block.attributes
		);
	} );
	// ...
}

// util.js
export function getHrefAndDestination(
	image,
	galleryDestination,
	imageDestination,
	attributes
) {
	// ...
	switch ( imageDestination ? imageDestination : galleryDestination ) {
		case LINK_DESTINATION_MEDIA_WP_CORE:
		case LINK_DESTINATION_MEDIA:
			return {
				href: image?.source_url || image?.url, // eslint-disable-line camelcase
				linkDestination: IMAGE_LINK_DESTINATION_MEDIA,
				lightbox: {
					...attributes?.lightbox,
					enabled: false,
				},
			};
		// ...
	}

	return {};
}
```